cli/src/pcluster/models/imagebuilder.py

```diff
@@ -0,0 +1,167 @@
+
+
+class Build(Resource):
+    """Represent the build configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        instance_type: str,
+        parent_image: str,
+        instance_role: str = None,  # TODO: auto generate if not assigned
+        subnet_id: str = None,  # TODO: auto generate if not assigned
+        tags: List[Tag] = None,
+        security_group_ids: List[str] = None,
+        components: List[Component] = None,
+    ):
+        super().__init__()
+        self.instance_type = instance_type
+        self.parent_image = parent_image
+        self.instance_role = instance_role
+        self.tags = tags
+        self.subnet_id = subnet_id
+        self.security_group_ids = security_group_ids
+        self.components = components
+        # TODO: add validator
+
+
+# ---------------------- Dev Settings ---------------------- #
+
+
+class ChefCookbook(Resource):
+    """Represent the chef cookbook configuration for the ImageBuilder."""
+
+    def __init__(self, url: str, json: str):
+        super().__init__()
+        self.url = url
+        self.json = json
+        # TODO: add validator
+
+
+class DevSettings(Resource):
+    """Represent the dev settings configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        update_os_and_reboot: bool = None,
+        disable_pcluster_component: bool = None,
+        chef_cookbook: ChefCookbook = None,
+        node_url: str = None,
+        aws_batch_cli_url: str = None,
+        distribution_configuration_arn: str = None,
+        terminate_instance_on_failure: bool = None,
+    ):
+        super().__init__()
+        self.update_os_and_reboot = update_os_and_reboot
+        self.disable_pcluster_component = disable_pcluster_component
+        self.chef_cookbook = chef_cookbook
+        self.node_url = node_url
+        self.aws_batch_cli_url = aws_batch_cli_url
+        self.distribution_configuration_arn = distribution_configuration_arn
+        self.terminate_instance_on_failure = terminate_instance_on_failure
+        self._set_default()
+        # TODO: add validator
+
+    def _set_default(self):
+        if self.update_os_and_reboot is None:
+            self.update_os_and_reboot = MarkedBool(False)
+        if self.disable_pcluster_component is None:
+            self.disable_pcluster_component = MarkedBool(False)
+        if self.terminate_instance_on_failure is None:
+            self.terminate_instance_on_failure = MarkedBool(True)
+
+
+# ---------------------- ImageBuilder ---------------------- #
+
+
+class ImageBuilder(Resource):
+    """Represent the configuration of an ImageBuilder."""
+
+    def __init__(
+        self,
+        image: Image,
+        build: Build,
+        dev_settings: DevSettings = None,
+    ):
+        super().__init__()
+        self.image = image
+        self.build = build
+        self.dev_settings = dev_settings
+        self._add_validator(BaseAMIValidator, ami_id=self.build.parent_image)
```

Comment:
The validators should be defined next to where the property is stored, i.e. in this case we are validating the `build.parent_image`, so the validator should be placed in the Build class.
Instead if the validator is checking a combination of properties from different sections, this is the right place where to define it